sheepdog/transactions/upload/entity.py

```diff
@@ -835,6 +855,41 @@ def set_association_proxies(self):
                         msg, keys=[name], type=EntityErrors.INVALID_LINK)
                     continue
 
+                # TT-273, check if we're doing sample->sample for restrictions
```

Comment:
Is this gdc-specific? If so, lets at least enclose it in `if IS_GDC`, which also is gross, true

Reply:
Consider this being an open-source repository, shouldn't put jira ticket number here which doesn't mean anything to non-gdc developers
Can you explain what is this code block trying to do?

Reply:
The idea was to restrict certain edges by what their parent type is. Sorry, this should totally have IS_GDC around it. I'll add.